grype/matcher/apk/matcher_test.go

```diff
@@ -70,42 +64,123 @@ func TestNoSecDBMatch(t *testing.T) {
 			must(syftPkg.NewCPE("cpe:2.3:a:*:libvncserver:0.9.9:*:*:*:*:*:*:*")),
 		},
 	}
-	matches, err := m.Match(provider, &d, p)
 
-	if err != nil {
-		t.Fatalf("failed to get matches: %+v", err)
+	vulnFound, err := vulnerability.NewVulnerability(secDbVuln)
+	assert.NoError(t, err)
+
+	expected := []match.Match{
+		{
+			Type:          match.ExactDirectMatch,
+			Confidence:    1.0,
+			Vulnerability: *vulnFound,
+			Package:       p,
+			SearchKey: map[string]interface{}{
+				"distro": map[string]string{
+					"type":    d.Type.String(),
+					"version": d.RawVersion,
+				},
+			},
+			SearchMatches: map[string]interface{}{
+				"constraint": vulnFound.Constraint.String(),
+			},
+			Matcher: match.ApkMatcher,
+		},
 	}
 
-	if len(matches) != 0 {
-		t.Errorf("expected 0 matches but got: %d", len(matches))
+	actual, err := m.Match(provider, &d, p)
+	assert.NoError(t, err)
+
+	for _, diff := range deep.Equal(expected, actual) {
+		t.Errorf("diff: %+v", diff)
 	}
 
 }
 
-func TestMatches(t *testing.T) {
-	// NVD and Alpine's secDB both have the same CVE ID for the package so it matches
+func TestBothSecdbAndNvdMatches(t *testing.T) {
```

Comment:
Related to this, but not quite the same the same test would be if the CVE ID is present in both NVD and SecDB, but the CPE match for NVD and package name for SecDB aren't the same (e.g. SecDB entry is for different package name). I think in that case, the logic will correctly remove the NVD match, but there isn't a test for that condition. Do you thin it warrants a specific test or the logic covered by these tests already?

Reply:
The nuance seems important given the difference in product-name and package-names we see, and this can be easily tested --I'll add a case for it 👍